modules/gapi/test/util/variant_tests.cpp

```diff
@@ -486,4 +500,92 @@ TEST(Variant, EXT_IndexOf)
     static_assert(6u == V::index_of<MyClass>(), "Index is incorrect");
 }
 
+namespace test_validation
+{
+struct MyType
+{
+    friend std::ostream& operator<<(std::ostream& out, const MyType& src)
+    {
+        return out << "MyType"; (void) src;
+    }
+};
+class MyClass
+{
+    friend std::ostream& operator<<(std::ostream& out, const MyClass& src)
+    {
+        return out << "MyClass"; (void) src;
+    }
+};
+
+struct MyParamVisitor : cv::util::static_visitor<bool, MyParamVisitor>
+{
+    MyParamVisitor(std::ostream &output) : out(output) {}
+
+    template<class Type>
+    bool visit(std::size_t index, Type val, int check)
+    {
+        bool result = false;
+        out << index << ":" << val <<",";
+        if(std::is_same<Type, int>::value)
+        {
+            result = !memcmp(&val, &check, sizeof(int));
+        }
+        return result;
+    }
+
+    std::ostream &out;
+};
+
+struct MyNoParamVisitor : cv::util::static_visitor<bool, MyNoParamVisitor>
+{
+    MyNoParamVisitor(std::ostream &output) : out(output) {}
+
+    template<class Type>
+    bool visit(std::size_t index, Type val)
+    {
+        out << index << ":" << val <<",";
+        return true;
+    }
+    std::ostream &out;
+};
```

Comment:
is outside of test scope because c++ told me 'local classes with templates are forbidden`
for special (non-templated) methods it should  compile fine